concent_api/core/models.py

```diff
@@ -684,3 +686,38 @@ class PendingEthereumTransaction(Model):
     s = DecimalField(max_digits=78, decimal_places=0)
 
     created_at = DateTimeField(auto_now_add=True)
+
+
+class DepositAccount(Model):
+    client = ForeignKey(Client)
+    ethereum_public_key = CharField(max_length=ETHEREUM_ADDRESS_LENGTH)
+    created_at = DateTimeField(auto_now_add=True)
+
+
+class DepositClaim(Model):
+    subtask = ForeignKey(Subtask, blank=True, null=True)
+    payer_deposit_account = ForeignKey(DepositAccount)
+    payee_ethereum_public_key = CharField(max_length=ETHEREUM_ADDRESS_LENGTH)
+    concent_use_case = IntegerField()
+    amount = DecimalField(max_digits=78, decimal_places=0)
+    tx_hash = CharField(max_length=64, blank=True, null=True, unique=True)
+    created_at = DateTimeField(auto_now_add=True)
+    modified_at = DateTimeField(auto_now=True)
+
+    def clean(self) -> None:
+        if self.subtask is None and self.concent_use_case != ConcentUseCase.ForcedPayment.name:
+            raise ValidationError({
+                'subtask: Can be NULL if and only if concent_use_case is ForcedPayment'
+            })
+        if not isinstance(DepositAccount, object) or self.payer_deposit_account.ethereum_public_key == self.payee_ethereum_public_key:
```

Comment:
`isinstance(DepositAccount, object)` this is wrong. You are checking if one **class** (type) has type of another class, whether you should check if particular **instance** has type of some class...
https://docs.python.org/3/library/functions.html#isinstance

Reply:
Done.

Reply:
What about this one?